source/extensions/filters/http/basic_auth/basic_auth_filter.h

```diff
@@ -36,15 +36,14 @@ struct User {
   std::string hash;
 };
 
-using UserMapConstPtr =
-    std::unique_ptr<const absl::flat_hash_map<std::string, User>>; // username, User
+using UserMap = absl::flat_hash_map<std::string, User>;
 
 /**
  * Configuration for the Basic Auth filter.
  */
 class FilterConfig {
 public:
-  FilterConfig(UserMapConstPtr users, const std::string& stats_prefix, Stats::Scope& scope);
+  FilterConfig(UserMap&& users, const std::string& stats_prefix, Stats::Scope& scope);
```

Comment:
There is always an interesting C++ feature to learn :-)
What is the difference between this way and a const pointer?  When to choose what?